GVFS/GVFS.Common/Git/GitAuthentication.cs

```diff
@@ -152,6 +164,28 @@ public bool TryInitializeAndRequireAuth(ITracer tracer, out string errorMessage)
             return false;
         }
 
+        private bool TryParseCachedAuthString(out string username, out string password)

```

Comment:
Instead of this I think we could just store the username/password in addition to the cachedAuthString.  I don't feel strongly.

Reply:
I did wonder about that, but then you've got to worry about keeping them in sync with each other.

Reply:
@mjcheetham which is exactly the other issue I was debating :-).

Reply:
I won't store the username/password as well as the basic auth string to avoid needing to keep them all in sync (and test for that). The deconstruction code here is only called once per mount process, and once per auth string, so it's not going to cause a noticeable perf problem to compute it rather than store the user/pass.